service/src/main/java/tennisclub/service/BookingServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package tennisclub.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import tennisclub.dao.BookingDao;
+import tennisclub.entity.Booking;
+import tennisclub.entity.Court;
+import tennisclub.entity.User;
+import tennisclub.exceptions.TennisClubManagerException;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class BookingServiceImpl implements BookingService {
+
+    final private BookingDao bookingDao;
+
+    @Autowired
+    public BookingServiceImpl(BookingDao bookingDao) {
+        this.bookingDao = bookingDao;
+    }
+
+    @Override
+    public void createBooking(Booking booking) {
+        bookingDao.create(booking);
+    }
+
+    @Override
+    public Booking updateBooking(Booking booking) {
+        return bookingDao.update(booking);
+    }
+
+    @Override
+    public void deleteBooking(Booking booking) {
+        bookingDao.delete(booking);
+    }
+
+    @Override
+    public void removeUser(Booking booking, User user) {
+        if (!booking.getUsers().contains(user)) {
+            throw new TennisClubManagerException("Removing user from a booking he is not in!");
+        }
+        booking.removeUser(user);
+    }
+
+    @Override
+    public void addUser(Booking booking, User user) {
+        if (booking.getUsers().contains(user)) {
+            throw new TennisClubManagerException("Adding user to a booking he is already in!");
+        }
+        booking.addUser(user);
```

Comment:
Shouldn't this actually be `user.addBooking()`? 🤔 Don't we need to set the booking at the user's side as well? 🤔

Reply:
Hmmm, maybe if hibernate sees a change in the owning side it also modifies the other side to reflect this? We would need to test this. But I guess using `user.addBooking()` would be safer since it sets both sides.